Tasks/Maven/CodeAnalysis/mavencodeanalysis.ts

```diff
@@ -67,28 +65,42 @@ export function uploadCodeAnalysisResults():void {
 }
 
 // Returns the names of any enabled code analysis tools, or empty array if none.
-function getEnabledCodeAnalysisTools():string[] {
-    var result:string[] = [];
+function getEnabledCodeAnalysisTools(): Set<string> {
+    var result: Set<string> = new Set<string>();
 
     if (tl.getBoolInput('pmdAnalysisEnabled', false)) {
-        console.log('PMD analysis is enabled');
-        result.push(pmd.toolName);
+        result.add(pmd.toolName);
+    }
+
+    if (tl.getBoolInput('sqAnalysisEnabled', false)) {
+        result.add(sq.toolName);
     }
 
     return result;
 }
 
 // Returns true if the given code analysis tool is enabled
-function isCodeAnalysisToolEnabled(toolName:string) {
+export function isCodeAnalysisToolEnabled(toolName:string): boolean {
     // Get the list of enabled tools, return whether or not toolName is contained in it
-    return (getEnabledCodeAnalysisTools().indexOf(toolName) > -1);
+    var result = getEnabledCodeAnalysisTools().has(toolName);
+    if (result) {
+        // Looks like: 'SonarQube analysis is enabled.'
+        console.log(tl.loc('codeAnalysis_toolIsEnabled', toolName));
```

Comment:
I think this is a good place to be logging it. Then we get a log message every time one of these tools finds it is enabled, and no message if they do not. Both are useful pieces of debug information, and moving the logging upwards would result in a lot of identical method calls.